### PR TITLE
Adjust for new container and testbed network specifics

### PR DESCRIPTION
**Pull request by reqa**
Adjustments to new container image and testbed network specifics

### CODE REVIEW

File: roles/ucs/defaults/main.yml [hunk 1]
```diff
@@ -9,24 +9,27 @@ operator_group: "{{ operator_user }}"
...
 ##########################
 # ucs
 
-ucs_hostname: ucs-master
-ucs_domainname: osism.local
-ucs_root_password: password
-
-ucs_configuration_directory: /opt/ucs/configuration
-ucs_docker_compose_directory: /opt/ucs
+ucs_configuration_directory: /opt/ucs-iam/configuration
```
Comment (berendt) [inline roles/ucs/defaults/main.yml:17]: Directory name == role name. Please do not change.

File: roles/ucs/defaults/main.yml [hunk 2]
```diff
@@ -9,24 +9,27 @@ operator_group: "{{ operator_user }}"
...
+ucs_docker_compose_directory: /opt/ucs-iam
```
Comment (berendt) [inline roles/ucs/defaults/main.yml:18]: Dito.

File: roles/ucs/defaults/main.yml [hunk 3]
```diff
@@ -9,24 +9,27 @@ operator_group: "{{ operator_user }}"
...
 
 ucs_host: 127.0.0.1
 ucs_port: 8011
 
-ucs_tag: latest
-ucs_image: "{{ docker_registry_ucs }}/univention/ucs-master-amd64:{{ ucs_tag }}"
+ucs_tag: ucs446
```
Comment (berendt) [inline roles/ucs/defaults/main.yml:23]: I would prefer to use the latest tag or a specific version at this point. ucs446 looks like a build tag?

File: roles/ucs/defaults/main.yml [hunk 4]
```diff
@@ -9,24 +9,27 @@ operator_group: "{{ operator_user }}"
...
+ucs_image: "{{ docker_registry_ucs }}/univention-corporate-server-scs:{{ ucs_tag }}"
 
-ucs_container_name: ucs
+ucs_container_name: ucs-iam
```
Comment (berendt) [inline roles/ucs/defaults/main.yml:26]: Do not change this.

File: roles/ucs/defaults/main.yml [hunk 5]
```diff
@@ -9,24 +9,27 @@ operator_group: "{{ operator_user }}"
...
 
 ucs_network: 172.31.100.128/28
+
+ucs_container_hostname: ucs-iam
```
Comment (berendt) [inline roles/ucs/defaults/main.yml:30]: Use ``ucs``.

File: roles/ucs/defaults/main.yml [hunk 6]
```diff
@@ -9,24 +9,27 @@ operator_group: "{{ operator_user }}"
...
+ucs_container_domainname: osism.local
+
+ucs_external_hostname: iam
```
Comment (berendt) [inline roles/ucs/defaults/main.yml:33]: Use ``ucs``.

File: roles/ucs/templates/docker-compose.yml.j2 [hunk 7]
```diff
@@ -4,29 +4,29 @@ version: '3'
...
     container_name: "{{ ucs_container_name }}"
-    hostname: "{{ ucs_hostname }}"
-    restart: unless-stopped
     image: "{{ ucs_image }}"
-    # NOTE(berendt): is unfortunately realized in the container image like this
-    command: /sbin/init
+    restart: unless-stopped
     env_file:
-      - "{{ ucs_configuration_directory }}/ucs.env"
+      - "{{ ucs_configuration_directory }}/ucs-iam.env"
```
Comment (berendt) [inline roles/ucs/templates/docker-compose.yml.j2:10]: Do not change this. Keep ``ucs.env`` as filename.

File: roles/ucs/templates/docker-compose.yml.j2 [hunk 8]
```diff
@@ -4,29 +4,29 @@ version: '3'
...
+    networks:
+      - ucs_default
 
 networks:
  default:
    driver: bridge
-   ipam:
-     driver: default
-     config:
-       - subnet:  {{ ucs_network }}
```
Comment (berendt) [inline roles/ucs/templates/docker-compose.yml.j2:32]: This is required. The CIDR within the services must be configurable.

File: roles/ucs/templates/docker-compose.yml.j2 [hunk 9]
```diff
@@ -4,29 +4,29 @@ version: '3'
...
       - /run
-      - /run/lock
+      - /tmp
     cap_add:
       - SYS_ADMIN
+      - MKNOD
     ports:
       - "{{ ucs_host }}:{{ ucs_port }}:80"
+    networks:
+      - ucs_default
```
Comment (berendt) [inline roles/ucs/templates/docker-compose.yml.j2:26]: This should not be necessary.

File: roles/ucs/defaults/main.yml [hunk 10]
```diff
@@ -9,24 +9,27 @@ operator_group: "{{ operator_user }}"
 # docker
 
 docker_registry: index.docker.io
-docker_registry_ucs: "{{ docker_registry }}"
+docker_registry_ucs: docker-test.software-univention.de
```
Comment (berendt) [inline roles/ucs/defaults/main.yml:12]: The image should come from a semi-official registry (Docker Hub, Quay, ..). Is this not published anywhere?

**Review by berendt — CHANGES_REQUESTED** [review]

File: roles/ucs/defaults/main.yml (hunk 10 above)
Comment (reqa) [reply roles/ucs/defaults/main.yml:12]: As the commit message suggests, this temporary solution required adjustments to the specifics of the testbed (MTU, Port-Number, HTTPS termination at traefik) which are not useful for other purposes. The source code of the container is published under https://github.com/univention/ucs-appliance-container and we will also publish that on docker hub soon. But for this project we currently need the adjustments specific for the testbed.

File: roles/ucs/defaults/main.yml (hunk 1 above)
Comment (reqa) [reply roles/ucs/defaults/main.yml:17]: Ok, thanks, I'll fix that and the corresponding points below.

**Comment by berendt** [comment]
Out of date.

**Pull request closed without merging**